src/Sulu/Component/Content/Document/Subscriber/DocumentSynchronizationSubscriber.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Content\Document\Subscriber;
+
+use Sulu\Component\Content\Document\Behavior\SynchronizeBehavior;
+use Sulu\Component\Content\Document\SynchronizationManager;
+use Sulu\Component\DocumentManager\Behavior\Mapping\LocaleBehavior;
+use Sulu\Component\DocumentManager\DocumentManagerInterface;
+use Sulu\Component\DocumentManager\Event\FlushEvent;
+use Sulu\Component\DocumentManager\Event\MetadataLoadEvent;
+use Sulu\Component\DocumentManager\Event\PersistEvent;
+use Sulu\Component\DocumentManager\Event\RemoveEvent;
+use Sulu\Component\DocumentManager\Events;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class DocumentSynchronizationSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var SynchronizationManager
+     */
+    private $syncManager;
+
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $defaultManager;
+
+    /**
+     * @var object[]
+     */
+    private $persistQueue = [];
+
+    /**
+     * @var object[]
+     */
+    private $removeQueue = [];
+
+    /**
+     * NOTE: We pass the default manager here because we need to ensure that we
+     *       only process documents FROM the default manager. If we could assign
+     *       event subscribers to specific document managers this would not
+     *       be necessary.
```

Comment:
Yes, it could now be removed.

On Thu, Apr 14, 2016 at 05:53:24AM -0700, Daniel Rotter wrote:
>    In
>    [1]src/Sulu/Component/Content/Document/Subscriber/DocumentSynchronizationSubscriber.php:
> 
>  > +
>  > +    /**
>  > +     * @var object[]
>  > +     */
>  > +    private $persistQueue = [];
>  > +
>  > +    /**
>  > +     * @var object[]
>  > +     */
>  > +    private $removeQueue = [];
>  > +
>  > +    /**
>  > +     * NOTE: We pass the default manager here because we need to ensure that we
>  > +     *       only process documents FROM the default manager. If we could assign
>  > +     *       event subscribers to specific document managers this would not
>  > +     *       be necessary.
> 
>    Isn't that something I have seen solved in the other PR?
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly or [2]view it on GitHub
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/dantleech/sulu/pull/2#discussion_r59711032
>    2. https://github.com/dantleech/sulu/pull/2/files/d181f9007b19b02c9fc7f5bcb5786b99f3f92c89#r59711032
>    3. https://github.com/dantleech/sulu/pull/2/files/d181f9007b19b02c9fc7f5bcb5786b99f3f92c89#r59711032